aodndata/nsw_oeh/handler.py

```diff
@@ -0,0 +1,432 @@
+            file_name = os.path.basename(file_path)
+            if not file_name:
+                continue  # skip directories
+
+            messages = self.check_name(file_name)
+
+            if file_name.endswith('ScientificRigour.pdf'):
+                have_metadata = True
+
+            if file_name.endswith('.xyz'):
+                have_xyz = True
+
+            # Check coverage shapefile
+            if file_name.endswith('_SHP.shp'):
+                have_coverage = True
+                messages.extend(self.check_shapefile('/' + file_path))
+
+            if messages:
+                report[file_name] = messages
+            else:
+                self.zip_contents.append(file_path)
+
+        # Overall checks...
+        messages = []
+
+        # metadata sheet (PDF) exists
+        if not have_metadata:
+            messages.append("Missing metadata file (PDF format)")
+
+        # shapefile exists
+        # TODO: Check there is exactly one coverage shapefile
+        if not have_coverage:
+            messages.append("Missing survey coverage shapefile")
+
+        # at least one XYZ file for MB surveys
+        if not have_xyz and self.survey_methods == 'MB':
+            messages.append("Missing bathymetry xyz file")
+
+        if messages:
+            report["Zip file contents"] = messages
+
+        return report
+
+    def get_dest_path(self):
+        """
+        Return the relative path where the contents of the zip file should be
+        published to on S3. Raise an exception if required metadata are missing.
+        :return: relative path
+        :rtype: str
+        """
+        survey_year = self.survey_name[:4]
+        methods_name = SURVEY_METHODS.get(self.survey_methods, '')
+        if not methods_name or not survey_year or not self.survey_name:
+            raise ValueError(
+                "Could not determine destination path for {zip}".format(zip=self.zip_file)
+            )
+        return os.path.join('NSW-OEH', methods_name, survey_year, self.survey_name)
+
+class NswOehHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(NswOehHandler, self).__init__(*args, **kwargs)
+
+        self.allowed_extensions = ['.zip']
+        self.survey_path = None
+
+    def preprocess(self):
+        self.file_collection.add(self.input_file_object)
+        self.input_file_object.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+
+        pz = NSWOEHSurveyProcesor(self.input_file)
+        report = pz.check_all()
+
+        if len(report) > 0:
+            raise_string = ''
+            for heading, messages in report.items():
+                raise_string += "\n{heading}:\n{messages}\n".format(
+                    heading=heading,
+                    messages="\n".join(["{0}. {1}".format(i+1, messages) for i, messages in enumerate(messages)]))
+
+            raise InvalidFileContentError(
+                "Could not determine data category for '{name}'\n{raise_string}".format(
```

Comment:
The error message should be something like
```suggestion
                "Zip file '{name}' failed content checks:'\n{raise_string}".format(
```